src/site/content/en/notifications/use-push-notifications-to-engage-users/index.md

```diff
@@ -0,0 +1,161 @@
+
+```js
+let title = 'Hi!';
+let options = { 
+  body: 'Very Important Message',
+  /* other options can go here */
+};
+let notification = new Notification(title, options);
+```
+
+The `title` is displayed in bold when the notification is active, while the `body` contains additional text.
+
+### Get permission to send notifications
+
+To display a notification, your app must get permission from the user to do so: 
+
+```js
+Notification.requestPermission();
+```
+
+{% Aside %}
+
+The web platform encourages developers to request notifications permission when users understand what is being asked and why. 
+
+Bring up the permission prompt **in response to a user gesture** and **when the value of notifications is obvious**. Avoid bringing up the permission prompt immediately when the user lands on the page.
+
+{% endAside %}
+
+## How do push notifications work?
+
+The real power of notifications comes from the combination of service workers and push technology.
+
+*   Service workers can run in the background and display notifications even when your app itself is not visible on screen. 
+
+*   Push technology lets a server decide when to send notifications. A push service creates unique URLs for each subscribed service worker. Sending messages to the URL raises events on the service worker.
+
+In the following example flow, the client and service worker use vanilla JavaScript with no extra libraries. The server is built with the [`express` npm package](https://www.npmjs.com/package/express) on [Node.js](https://nodejs.org/en/), and uses the [`web-push` npm package](https://www.npmjs.com/package/web-push) to send notifications. To send information to the server, the client makes a call to a POST URL that the server has exposed.
+
+### Part 1: Registration and subscription
+
+1.  A client app registers a service worker with `ServiceWorkerContainer.register()`. The registered service worker will continue to run in the background when the client is inactive. 
+
+    Client code: 
+
+    ```js
+    navigator.serviceWorker.register('sw.js');
+    ```
+
+1.  The client requests permission from the user to send notifications.
+
+    Client code: 
+
+    ```js
+    Notification.requestPermission();
+    ```
+
+1.  To enable push notifications, the service worker uses `PushManager.subscribe()` to create a push subscription. In the call to `PushManager.subscribe()`, the service worker supplies the app's API key as an identifier.
+
+    Client code:
+
+    ```js/0-1/0/
+    navigator.serviceWorker.register('sw.js').then(sw => {
+      sw.pushManager.subscribe({ /* API key */ });
+    });
+    ```
+
+    Push services like Firebase Cloud Messaging operate on a subscription model. When a service worker subscribes to a push service by calling `PushManager.subscribe()`, the push service creates a URL unique to that service worker. The URL is known as a subscription endpoint. 
```

Comment:
```suggestion
    Push services like Firebase Cloud Messaging operate on a subscription model. When a service worker subscribes to a push service by calling `PushManager.subscribe()`, the push service creates a URL unique to that service worker. The URL is known as a **subscription endpoint**. 
```
Proposing emphasis since this new term is taken up in the next step.